corehq/messaging/management/commands/manage_rule_run.py

```diff
@@ -0,0 +1,84 @@
+from corehq.apps.data_interfaces.models import AutomaticUpdateRule
+from corehq.messaging.util import MessagingRuleProgressHelper
+from django.core.management.base import BaseCommand, CommandError
+
+
+class Command(BaseCommand):
+    help = """
+        Check status of and cancel currently-running AutomaticUpdateRule runs.
+        If domain and rule-id are given, will check on and optionally cancel that rule.
+        If no arguments are passed, will display status of all currently locked rules.
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--domain',
+            help='Required if passing rule-id. Acts as a sanity check while fetching given rule.'
+        )
+        parser.add_argument(
+            '--rule-id',
+            help='If provided, fetch status for only this rule',
+        )
+        parser.add_argument(
+            '--cancel',
+            action='store_true',
+            default=False,
+            help='Cancel the given rule',
+        )
+
+    def get_rule(self, domain, rule_id):
+        try:
+            rule = AutomaticUpdateRule.objects.get(pk=rule_id)
+        except AutomaticUpdateRule.DoesNotExist:
+            raise CommandError("Rule not found")
+
+        if rule.domain != domain:
+            raise CommandError("Domain '%s' does not match rule's domain '%s'" % (domain, rule.domain))
+
+        if rule.workflow != AutomaticUpdateRule.WORKFLOW_SCHEDULING:
+            raise CommandError("Expected the rule to be a messaging rule")
+
+        return rule
+
+    def print_status(self, rule):
+        rule = AutomaticUpdateRule.objects.get(id=rule.id)
```

Comment:
why did we fetch it again with id now?

Reply:
Oops, I copied this from another script. Will fix.